_posts/2017-03-15-Performance-of-LSTM-Using-CuDNN-v5.md

```diff
@@ -0,0 +1,135 @@
+
+Comparing the setting with/without cuDNN, about 2 times ~ 2.8 times faster when using cuDNN in forward time, and 1.6 times faster in backward time.
+
+ 
+![The effect of mini-batch size]({{ site.url }}/assets/nsteplstm-cudnn-v5-minibatch.png)
+    
+
+* The effect of the layer size of LSTM and sequence length of data
+parameters: length={5, 25, 50}, layer={1, 2, 3}
+In the setting without cuDNN, as the length of data and the number of layers increases, it takes longer time than the setting with cuDNN.
+
+When we use a large LSTM setting (layer=3, length=50), about 7 times faster in forward time, and 4 times faster in backward time with cuDNN.
+
+When we use a small LSTM setting (layer=1, length=5), about 1.5 times faster in forward time, and 1.04 times faster in backward time.
+
+
+![The effect of the layer size of LSTM and sequence length of data]({{ site.url }}/assets/nsteplstm-cudnn-v5-length-layers.png)
+
+* The effect of the layer size of LSTM and random sequence length of data
+parameters: layer = {1, 2, 3}, random={True, False}
+In this setting, we compare if the data sequence length is fixed or not. (Max length is 25.)
+When we use cuDNN, the effect of randomness is small. 
+
+![The effect of the layer size of LSTM and random sequence length of data]({{ site.url }}/assets/nsteplstm-cudnn-v5-random-seq.png)
+
+* The effect of the layer size of LSTM and the input unit size
+parameters:  layer = {1, 2, 3}, input={128, 256}
+In the setting with cuDNN, as number of layers increases, the difference between cuDNN and no-cuDNN will be large. (About 5 times faster in forward time, and about 2.7 times faster in backward time.)
+
+
+![The effect of the layer size of LSTM and the input unit size]({{ site.url }}/assets/nsteplstm-cudnn-v5-input.png)
+
+
+
+* The effect of the layer size of LSTM and the hidden unit size
+parameters: layer = {1, 2, 3}, hidden={128, 256}
+In the setting with cuDNN, the number of layers increases, the difference between cuDNN and no-cuDNN will be large. (This is same as layer size and input size experiment.)
+However as hidden unit size increases, the difference between cuDNN and no-cuDNN will be small. 
+
+![The effect of the layer size of LSTM and the hidden unit size]({{ site.url }}/assets/nsteplstm-cudnn-v5-hidden.png)
+
+
+* The effect of the layer size of LSTM and dropout rate
+parameters: layer={1, 2, 3}, dropout={0.5, 0.0}
+In the setting with cuDNN, when using dropout, the speed gets slower but the difference is very small (dropout rate=0.50).
+
+![The effect of the layer size of LSTM and dropout rate]({{ site.url }}/assets/nsteplstm-cudnn-v5-drop.png)
+
+
+
+* When will the differences be large? (Setting with/without cuDNN.)
+
+As batch size is small (batchsize=128), sequence length is long (length=50) and the number of layers is large (layer=3), the difference is large. (The cuDNN is faster than no-cuDNN setting.)
+
+If we use a large LSTM in the experiment, the effect of cuDNN will be large.
```

Comment:
Consider rewording:
If we use a large LSTM in the experiment, the performance benefit of using cuDNN will be large.